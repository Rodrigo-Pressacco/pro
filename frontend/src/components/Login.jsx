import React, { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import axios from 'axios';
import "../styles/Login.css";
import url from "../utils/datosInicio.json"

const Login = (props) => {
    /*mandar email*/


   

    /*fin mandar email*/
    const [loginEmail, setLoginEmail] = useState("");
    const [loginPassword, setLoginPassword] = useState("");
    const [loginErrorEmail, setLoginErrorEmail] = useState();
    const [loginErrorPassword, setLoginErrorPaswword] = useState();
    const [loginButton, setLoginButton] = useState(false);

    const [elementoEnviado, setElementoEnviado] = useState();
    const localStorageKey = "myKey";
    const localStorageKeyToken = "myKeyForToken";
    const localStorageKeyPersona = "myKeyForPersona";
    const localStorageLogueado = "logueado";

    const [datosUsuarioTraidos, setdatosUsuarioTraidos] = useState([{
        id: null,
        email: null
    }]);

    useEffect(() => {
        if (datosUsuarioTraidos[0].email === null) {
            axios.get(`http://${url.hostUrl}:8080/usuarios/todos`)
                .then((response) => {
                    setdatosUsuarioTraidos(response.data)
                })
        }
    });

    const arrayFiltradoEmail = datosUsuarioTraidos.filter(element => {
        return element.email === loginEmail
    })


    const handleChangeEmail = e => {
        setLoginEmail(e.target.value);
    }

    const handleChangePassword = e => {
        setLoginPassword(e.target.value);
    }

    const blurEmail = e => {
        if (arrayFiltradoEmail.length !== 0) {
            setLoginErrorPaswword(false)
            setLoginErrorEmail(false)
        }
        else {
            setLoginButton(false)
            setLoginErrorEmail(true)
        }
    }

    const blurPassword = () => {
        if (arrayFiltradoEmail[0] !== null) {
            setLoginButton(true)

            setLoginErrorPaswword(false)
            setLoginErrorEmail(false)
            setElementoEnviado(arrayFiltradoEmail[0])

        }
        else {
            setLoginButton(false)
            setLoginErrorPaswword(true)
        }
    }

    function handleClickButton(e) {
        e.preventDefault()
    }

    function onSubmitForm() {
        
        props.logeado(true)

        axios.post(`http://${url.hostUrl}:8080/authenticate`, {
            username: elementoEnviado.email,
            password: loginPassword
        }, {
            headers: {
                'accept': '*/*',
                'content-type': 'application/json'
            }
        })
            .then((response) => {
                localStorage.setItem(localStorageKey, JSON.stringify(elementoEnviado))
                localStorage.setItem(localStorageKeyPersona, JSON.stringify(response.data[0]))
                localStorage.setItem(localStorageKeyToken, JSON.stringify(response.data[1]))
                localStorage.setItem(localStorageLogueado, JSON.stringify(true))
                props.dataUsuario(response.data[0])
                
            })
            .catch((error) => {
                console.log("hola");
                console.log(error);
            });
    }

    const cambiarBotenes = (datos) => {
        props.cambiarBotton(datos)
    }

    return (
        <div className="div-login">
            <h1>Ingresar</h1>
            <form className="form-Login">
                <div className="div-email-password-login">
                    {loginErrorEmail || loginErrorPassword ? <p>Por favor, vuelva a intentarlo sus credenciales son inválidas</p> : <></>}
                    <label>Email:</label>
                    <input type="email" name="login-email" id="login-email" className="login-email-password" onChange={handleChangeEmail} onBlur={blurEmail} />
                </div>
                <div className="div-email-password-login">
                    <label>Contraseña:</label>
                    <input type="password" name="login-password" id="login-password" className="login-email-password" onChange={handleChangePassword} onBlur={blurPassword} />
                </div>
                <div className="div-button">
                    {loginButton ? <Link to="/" className="login-link" onClick={onSubmitForm}>Ingresar</Link> :
                        <button onClick={handleClickButton} className="login-link login-button">Ingresar</button>
                    }

                    <p>¿Aún no tenes cuenta? <Link to="/registrarme" onClick={() => cambiarBotenes(2)} >Registrate</Link></p>
                </div>
            </form>
        </div>
    )
}

export default Login;