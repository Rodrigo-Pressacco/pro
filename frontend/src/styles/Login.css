.div-login{
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 138px;
    height: 570px;
}

.div-login h1{
    font-style: normal;
    font-weight: bold;
    font-size: 24px;
    line-height: 28px;
    text-align: center;
    color: var(--main-color);
}

.form-Login{    
    display: flex;
    flex-direction: column;
    width: 422px;
}

.div-email-password-login{
    display: flex;
    flex-direction: column;
    margin: 18px 0px;
}

.login-email-password{    
    height: 40px;
    padding-left: 13px;
    background: #FFFFFF;
    box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.15);
    border-radius: 5px;
    border: none;
}

.div-email-password-login label{
    height: 22px;
}

.login-email-error{    
    height: 40px;
    padding-left: 13px;
    background: #FFE1E1;
    border: 1px solid #FF0000;
    box-sizing: border-box;
    box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.15);
    border-radius: 5px;
}

.div-email-password-login p{
    width: 100%;
    height: 40px;
    background-color: #ffdada;
    border-radius: 5px;
    border: 1px #FF0000 solid;
    font-style: normal;
    font-weight: 500;
    font-size: 12px;
    line-height: 14px;
    text-align: center;
    color: #FF0000;
    padding: 14px;
    margin-bottom: 10px;
}

.login-email-password,
.div-email-password-login label{
    font-style: normal;
    font-weight: 500;
    font-size: 12px;
    line-height: 14px;
    color: var(--secondary-color);
}

.login-link{
    background: var(--main-color);
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.12);
    border-radius: 5px;
    border: none;
    height: 40px;
    width: 46%;
    margin-bottom: 15px;    
    padding-top: 12px;

    font-style: normal;
    font-weight: bold;
    font-size: 16px;
    line-height: 19px;
    text-align: center;
    color: #FFFFFF;
    text-decoration: none;
}

.login-button{
    padding-bottom: 10px;
}

@media (max-width: 777px){
    .form-Login{
        width: 346px;
        height: 728px;
    }
    .login-link{
        width: 100%;
    }
}

@media (max-width: 485px){
.login-link{
    width: 100%;
    }
}