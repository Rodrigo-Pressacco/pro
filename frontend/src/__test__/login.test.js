import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import Login from '../components/Login'

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('<Login />', () => {
    let login
    let handleChangeEmail = jest.fn(); 
    let handleChangePassword = jest.fn(); 
    let blurEmail = jest.fn(); 
    let blurPassword = jest.fn();
    let handleClickButton = jest.fn();
    let onSubmitForm = jest.fn();
    let cambiarBotenes = jest.fn();
    
    beforeEach(() => {
        login = shallow(<Login  handleChangeEmail={handleChangeEmail} 
                                handleChangePassword={handleChangePassword} 
                                blurEmail={blurEmail}
                                blurPassword={blurPassword}
                                handleClickButton={handleClickButton}
                                onSubmitForm={onSubmitForm}
                                cambiarBotenes={cambiarBotenes} />)
    });

    it("Deberia mostrar <Login/> correctamente", () => {
        expect(login).toMatchSnapshot();
    });

})