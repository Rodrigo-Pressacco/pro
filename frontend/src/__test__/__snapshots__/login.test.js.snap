// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Deberia mostrar <Login/> correctamente 1`] = `
<div
  className="div-login"
>
  <h1>
    Ingresar
  </h1>
  <form
    className="form-Login"
  >
    <div
      className="div-email-password-login"
    >
      <label>
        Email:
      </label>
      <input
        className="login-email-password"
        id="login-email"
        name="login-email"
        onBlur={[Function]}
        onChange={[Function]}
        type="email"
      />
    </div>
    <div
      className="div-email-password-login"
    >
      <label>
        Contraseña:
      </label>
      <input
        className="login-email-password"
        id="login-password"
        name="login-password"
        onBlur={[Function]}
        onChange={[Function]}
        type="password"
      />
    </div>
    <div
      className="div-button"
    >
      <button
        className="login-link login-button"
        onClick={[Function]}
      >
        Ingresar
      </button>
      <p>
        ¿Aún no tenes cuenta? 
        <Link
          onClick={[Function]}
          to="/registrarme"
        >
          Registrate
        </Link>
      </p>
    </div>
  </form>
</div>
`;
